metasys-server/openapi-snippet
# Revise parameter handling for snippets

This commit does several related things.

* consider examples for parameter values

  If `example` isn't set for a parameter, then go ahead
  and check for `examples`. If that is present, then use the
  value from the first entry as the parameter value.

  Fix #81

* implement support for explode for array and object

  Fix #83

* Fix 3 test scenarios that showed unexploded form
  parameters when explode should have default to true

  This was documented in #83

* Resolve $ref nodes for header parameters

  Fix #84

* Add a full suite of tests for parameters.

* Add support for cookie parameters

* Handle the merging of path and operation parameters correctly for
  all parameter types, not just query ones.

* Add support for label and matrix style path parameters

* Add support for space and pipe delimited query parameters

* There is support for deepObject style, but the `httpsnippet` library 
  encodes all the special chars so I don't know if that's correct or not.
